internal/pkg/services/argus/utils/utils.go

```diff
@@ -56,6 +57,143 @@ func LoadPlanId(planName string, resp *argus.PlansResponse) (*string, error) {
 	}
 }
 
+func MapToUpdateScrapeConfigPayload(resp *argus.GetScrapeConfigResponse) (*argus.UpdateScrapeConfigPayload, error) {
+	if resp == nil || resp.Data == nil {
+		return nil, fmt.Errorf("no Argus scrape config provided")
+	}
+
+	data := resp.Data
+
+	basicAuth := mapBasicAuth(data.BasicAuth)
+
+	var staticConfigs *[]argus.UpdateScrapeConfigPayloadStaticConfigsInner
+	if data.StaticConfigs != nil {
+		configs := make([]argus.UpdateScrapeConfigPayloadStaticConfigsInner, 0)
+		for _, config := range *data.StaticConfigs {
+			newConfig, err := mapStaticConfig(config)
+			if err != nil {
+				return nil, fmt.Errorf("map static config: %w", err)
+			}
+			configs = append(configs, newConfig)
+		}
+		staticConfigs = &configs
+	}
+
+	tlsConfig := mapTlsConfig(data.TlsConfig)
+
+	var metricsRelabelConfigs *[]argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner
+	if data.MetricsRelabelConfigs != nil {
+		configs := make([]argus.CreateScrapeConfigPayloadMetricsRelabelConfigsInner, 0)
+		for _, config := range *data.MetricsRelabelConfigs {
+			configs = append(configs, mapMetricsRelabelConfig(config))
+		}
+		metricsRelabelConfigs = &configs
+	}
+
+	var params *map[string]interface{}
+
+	if data.Params != nil {
+		paramsMap, err := convertMapAnyToInterface(*data.Params)
+		if err != nil {
+			return nil, fmt.Errorf("convert params: %w", err)
+		}
+		params = &paramsMap
+	}
+
+	return &argus.UpdateScrapeConfigPayload{
+		BasicAuth:             basicAuth,
+		BearerToken:           data.BearerToken,
+		HonorLabels:           data.HonorLabels,
+		HonorTimeStamps:       data.HonorTimeStamps,
+		MetricsPath:           data.MetricsPath,
+		MetricsRelabelConfigs: metricsRelabelConfigs,
+		Params:                params,
+		SampleLimit:           convertIntToFloat64(data.SampleLimit),
+		Scheme:                data.Scheme,
+		ScrapeInterval:        data.ScrapeInterval,
+		ScrapeTimeout:         data.ScrapeTimeout,
+		StaticConfigs:         staticConfigs,
+		TlsConfig:             tlsConfig,
+	}, nil
+}
+
+func convertIntToFloat64(i *int64) *float64 {
```

Comment:
Quite unfortunate that the API has different types in the response and in the payload.

I think this fits better in the general `utils` pkg, might be useful for other services in the future.

Reply:
also, add unit test for it